### PR TITLE
feat: add getIssuerForBrand to zoe

### LINKED ISSUE #1270: Hooks need the issuer for the amounts in the offer
**What is the Problem Being Solved?**

Offer hooks in some contracts need to be able to get the issuer for operations like `burn`.

**Description of the Design**

Add a `zcf.getIssuer(brand)` operation so that the issuer corresponding to the brand of the amount, and therefore to the corresponding payment, can be retrieved.

**Security Considerations**
None.

**Test Plan**
Add unit tests for that operation.

### PR DESCRIPTION
**Pull request by dtribble**
closes: #1270 

This adds `getIssuerForBrand` to `zcf` so that contracts can synchronously get issuers for amounts they receive. (We could add one to get it from an `Amount` as well.) The also adds type declaration.

This also modifies multipool `autoswap` to return the issuers etc. when a pool is added, rather than jut an acknowledgement string. 

# Testing

There is no existing test for `getBrandForIssuer`.  The test for this new operation should be added with that. The operation currently gets ad hoc testing because the subsystem tests or dapp-token-economy relies on it.

The tests for multipool autoswap were updated are updated to accomodate and test that the new return result returns a record with at least some expected fields.

# Documentation

This adds a new operation to ZCF, which should be added to the API documentation site.

This changes the return result of creating a multipool autoswap. The previous return was just a worthless string. Now we can get the issuers, etc. That's not currently documented, but could be extended with that information.

### CODE REVIEW

**Comment by katelynsills** [comment]
This looks good. Before merging can we add a unit test and coordinate with @tyg to change the documentation?

**Review by katelynsills — CHANGES_REQUESTED** [review]
I think there are some linting errors and we might need to rebase this on master for only the wanted changes to show. Can we also add unit tests for this and coordinate with Tom on updating the documentation?

**Comment by katelynsills** [comment]
Added a test, @dtribble do you want to take a look at it? I also asked Chris to review

**Review by katelynsills — APPROVED** [review]

File: packages/zoe/test/unitTests/contracts/zcfTesterContract.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    assert.equal(
+      zcf.getIssuerForBrand(brandKeywordRecord[keyword]),
+      issuerKeywordRecord[keyword],
+    );
+  });
+  return zcf.makeInvitation(() => {}, 'test');
+};
+
+harden(makeContract);
+export { makeContract };
```
Comment (Chris-Hibbert) [inline packages/zoe/test/unitTests/contracts/zcfTesterContract.js:26]: Is this intended to be called? Is it a replacement for current functionality in the tests?

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]
These look like useful helpers. I'm skeptical about adding a helper for the tests and not invoking it anywhere.

**Comment by katelynsills** [comment]
> These look like useful helpers. I'm skeptical about adding a helper for the tests and not invoking it anywhere.

I'm not sure what you mean. `zcfTesterContract.js` is run as the main part of this test, so everything added is invoked

**Comment by Chris-Hibbert** [comment]
> `zcfTesterContract.js` is run as the main part of this test, so everything added is invoked

Thanks, I was looking for `makeContract` and missed that  the file was imported explicitly.

**Review by Chris-Hibbert — APPROVED** [review]

**Pull request merged**